ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -1242,6 +1242,65 @@ keys:
                     type: str
                     required: true
                     description: Virtual IPv6 address
+        switchport:
+          type: dict
+          keys:
+            port_security:
+              type: dict
+              keys:
+                enabled:
+                  type: bool
+                mac_address_maximum:
+                  type: dict
+                  description: Maximum number of MAC addresses allowed on the interface.
+                  keys:
+                    disabled:
+                      type: bool
+                      description: Disable port level check for port security (only in shutdown mode).
+                    limit:
+                      type: int
+                      convert_types:
+                        - str
+                      min: 1
+                      max: 1000
+                      description: MAC address limit.
+                violation:
+                  type: dict
+                  description: Configure violation mode (shutdown or protect), default is 'shutdown'.
+                  keys:
+                    protect:
+                      type: dict
+                      description: Configure port security in protect mode.
+                      keys:
+                        enabled:
+                          type: bool
+                        log:
+                          type: bool
+                          description: Log new addresses seen after limit is reached in protect mode.
+                vlan:
```

Comment:
I think it is possible to have multiple ranges of vlans so I suggest modifying the data model to:
```yaml
...
port_security:
  ...
  vlan_default_mac_address_maximum: <int>
  vlans:
    - range: <str>
      mac_address_maximum: <int>
```
will that work?

Reply:
yes, it can be done

Reply:
Modified